lib/Doctrine/ODM/PHPCR/Proxy/ProxyFactory.php

```diff
@@ -29,6 +28,7 @@
 use Doctrine\Common\Util\ClassUtils;
 use Doctrine\ODM\PHPCR\DocumentManagerInterface;
 use Doctrine\ODM\PHPCR\Mapping\ClassMetadata;
+use Doctrine\Persistence\Mapping\ClassMetadata as BaseClassMetadata;
```

Comment:
i wonder if this really works with php < 7.4.

we use this class in the skipClass method and that method is defined by the base class with the common persistence class metadata. phpstorm tells me that restricting the class will only work from php 7.4 on

Reply:
I don't understand what you mean by "restricting the class", can you please elaborate?

Reply:
this factory extends a base class that declares the skipClass method but with Doctrine\Common\Persistence\Mapping\ClassMetadata as argument.

i thought PHP 7.4 allows this, but according to https://3v4l.org/rCbc9 all PHP 7 emit a warning, while PHP 8 will fatal on this.